tests/unit/test_snap.py

```diff
@@ -551,26 +551,36 @@ def raise_error(cmd, **kwargs):
     def test_snap_set(self, mock_subprocess):
         foo = snap.Snap("foo", snap.SnapState.Present, "stable", "1", "classic")
 
-        foo.set({"bar": "baz"})
+        configs = {
+            "true": True,
+            "false": False,
+            "null": None,
+            "integer": 1,
+            "float": 2.0,
+            "list": [1, 2.0, True, False],
+            "dict": {
+                "true": True,
+                "false": False,
+                "null": None,
+                "integer": 1,
+                "float": 2.0,
+                "list": [1, 2.0, True, False],
+            },
+        }
+
+        foo.set(configs, use_json=True)
+        args = ["{}={}".format(key, json.dumps(val)) for key, val in configs.items()]
         mock_subprocess.assert_called_with(
-            ["snap", "set", "foo", 'bar="baz"'],
+            ["snap", "set", "foo", *args],
```

Comment:
I would avoid copying the `json.dumps` and the list comprehension into the test, and (especially with the above simplification) just include the expected output verbatim. Something like:

```python
mock_subprocess.assert_called_with(
            ["snap", "set", "foo", "-t", 'n=42', 's="string"', 'd={"nested": True}'],
)
```

Reply:
Sure